packages/insomnia-components/components/sidebar.js

```diff
@@ -140,204 +213,637 @@ const StyledPanel: React.ComponentType<{}> = styled(motion.div)`
+  const handleServersVisibleClick = e => {
+    e.stopPropagation();
+    setServersVisible(!serversVisible);
+  };
+
+  const [requestsVisible, setRequestsVisible] = useState(true);
+  const handleRequestsVisibleClick = e => {
+    e.stopPropagation();
+    setRequestsVisible(!requestsVisible);
+  };
+
+  const [responsesVisible, setResponsesVisible] = useState(true);
+  const handleResponsesVisibleClick = e => {
+    e.stopPropagation();
+    setResponsesVisible(!responsesVisible);
+  };
+
+  const [parametersVisible, setParametersVisible] = useState(true);
+  const handleParametersVisibleClick = e => {
+    e.stopPropagation();
+    setParametersVisible(!parametersVisible);
+  };
+
+  const [headersVisible, setHeadersVisible] = useState(true);
+  const handleHeadersVisibleClick = e => {
+    e.stopPropagation();
+    setHeadersVisible(!headersVisible);
+  };
+
+  const [schemasVisible, setSchemasVisible] = useState(true);
+  const handleSchemasVisibleClick = e => {
+    e.stopPropagation();
+    setSchemasVisible(!schemasVisible);
+  };
+
+  const [securityVisible, setSecurityVisible] = useState(true);
+  const handleSecurityVisibleClick = e => {
+    e.stopPropagation();
+    setSecurityVisible(!securityVisible);
+  };
+
+  // Sections
+  if (props.jsonData === null) {
+    return null;
+  }
+  const { requestBodies, responses, parameters, headers } = props.jsonData.components;
+  const { servers } = props.jsonData;
+  const paths = Object.entries(props.jsonData.paths);
 
   return (
     <StyledSidebar className="theme--sidebar">
       <StyledSection>
         <StyledHeader>
-          <h6>INFO</h6>
-        </StyledHeader>
-      </StyledSection>
-
-      <StyledSection>
-        <StyledHeader>
-          <h6 onClick={toggleVisible2}>SERVERS</h6>
+          <h6 onClick={toggleInfoSec}>INFO</h6>
           <div>
-            <motion.span
-              initial={{ opacity: visible2 ? 0.6 : 0 }}
-              animate={{ opacity: visible2 ? 0.6 : 0 }}
-              transition={{ duration: 0.2, ease: 'easeInOut', delay: 0.2 }}>
-              <SvgIcon icon={IconEnum.trashcan} />
-            </motion.span>
-            <motion.span
-              onClick={toggleVisible3}
-              initial={{ opacity: visible2 ? 0.6 : 0 }}
-              animate={{ opacity: visible2 ? 0.6 : 0 }}
-              transition={{ duration: 0.2, ease: 'easeInOut', delay: 0.15 }}>
-              <SvgIcon icon={IconEnum.search} />
-            </motion.span>
-            <motion.span
-              initial={{ opacity: visible2 ? 0.6 : 0.4 }}
-              animate={{ opacity: visible2 ? 0.6 : 0.4 }}
-              transition={{ duration: 0.2, ease: 'easeInOut', delay: 0.1 }}>
-              <Tooltip message="Add" position="top">
-                <SvgIcon icon={IconEnum.plus} />
-              </Tooltip>
-            </motion.span>
+            <Dropdown renderButton={({ open }) => <SvgIcon icon={IconEnum.ellipsesCircle} />}>
```

Comment:
I believe this syntax will result in this drop-down component and it's children updating on every render. The inline function will mean a new instance is sent every time. Definitely a candidate to declare the a button component outside the component, and consume it here. Also, looks like the `open` prop isn't being used so I have removed it. Eg:
```
const DropdownElipsis = () => <SvgIcon icon={IconEnum.ellipsesCircle} />;
```

```suggestion
            <Dropdown renderButton={DropdownElipsis}>
```